# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsLineLayout.h
# Commit: eb0b125d3214
# Full Hash: eb0b125d3214a04c293cbe2f21184e85ae00302a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-09-28 21:51:27
# Regressor Bug: 1731541
# File Overlap Count: 1
# Description:
#   Bug 1731541 - Implement text-wrap: balance for nsBlockFrame reflow. r=emilio
#   
#   A simple form of balance for short blocks, implemented by incrementally reducing
#   the effective inline-size used during line-breaking, up to the point where an
#   extra line would be created.
# ==============================================================================

diff -r af42a433722a -r eb0b125d3214 layout/generic/nsLineLayout.h
--- a/layout/generic/nsLineLayout.h	Thu Sep 28 17:28:18 2023 +0000
+++ b/layout/generic/nsLineLayout.h	Thu Sep 28 17:28:19 2023 +0000
@@ -48,7 +48,10 @@
   void BeginLineReflow(nscoord aICoord, nscoord aBCoord, nscoord aISize,
                        nscoord aBSize, bool aImpactedByFloats,
                        bool aIsTopOfPage, mozilla::WritingMode aWritingMode,
-                       const nsSize& aContainerSize);
+                       const nsSize& aContainerSize,
+                       // aInset is used during text-wrap:balance to reduce
+                       // the effective available space on the line.
+                       nscoord aInset = 0);
 
   void EndLineReflow();
 
@@ -494,6 +497,7 @@
     nscoord mIStart;
     nscoord mICoord;
     nscoord mIEnd;
+    nscoord mInset;
 
     nscoord mBStartLeading, mBEndLeading;
     nscoord mLogicalBSize;